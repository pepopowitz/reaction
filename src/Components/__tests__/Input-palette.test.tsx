import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import Input from "../Input-palette"

describe("Input", () => {
  it("renders standard mode", () => {
    const input = renderer.create(
      <Input placeholder="Placeholder" title="Title" block />
    )

    expect(input.toJSON()).toMatchSnapshot()
  })

  it("renders quick mode", () => {
    const input = renderer.create(
      <Input quick placeholder="Placeholder" title="Title" block />
    )

    expect(input.toJSON()).toMatchSnapshot()
  })
})
