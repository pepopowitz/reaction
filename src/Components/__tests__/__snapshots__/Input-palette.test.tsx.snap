// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders quick mode 1`] = `
.c0 {
  padding-bottom: 5px;
}

.c3 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  border: 0;
  font-size: 17px;
  outline: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  padding: 0 10px;
  line-height: initial;
}

.c1 {
  padding: 10px;
  box-shadow: none;
  -webkit-transition: border-color 0.25s;
  transition: border-color 0.25s;
  margin-right: 10px;
  resize: none;
  display: block;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  border: 1px solid #e5e5e5;
  -webkit-transition: border-color 0.25s;
  transition: border-color 0.25s;
  margin-right: 0;
  margin-top: 5px;
  margin-bottom: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 20px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: content-box;
}

.c1::-webkit-input-placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c1::-moz-placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c1:-ms-input-placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c1::placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c1:hover,
.c1:focus,
.c1.focused {
  border-color: #6e1fff;
  outline: 0;
}

.c1:disabled {
  border: 2px dotted #e5e5e5;
}

.c2 {
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 12px;
  line-height: 1.4em;
  position: absolute;
  left: 10px;
  top: 7px;
  visibility: hidden;
  -webkit-animation: hSsjZP 0.2s linear;
  animation: hSsjZP 0.2s linear;
  -webkit-transition: visibility 0.2s linear;
  transition: visibility 0.2s linear;
  z-index: 1;
}

.c4 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 12px;
  line-height: 1.4em;
  margin-top: 0;
  color: #f7625a;
  visibility: hidden;
  -webkit-transition: visibility 0.2s linear;
  transition: visibility 0.2s linear;
  -webkit-animation: 0.25s linear;
  animation: 0.25s linear;
  height: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    />
    <input
      className="c3"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Placeholder"
      title="Title"
      value=""
    />
  </div>
  
  <div
    className="c4"
  />
</div>
`;

exports[`Input renders standard mode 1`] = `
.c0 {
  padding-bottom: 5px;
}

.c2 {
  padding: 10px;
  box-shadow: none;
  -webkit-transition: border-color 0.25s;
  transition: border-color 0.25s;
  margin-right: 10px;
  resize: none;
  display: block;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  border: 1px solid #e5e5e5;
  -webkit-transition: border-color 0.25s;
  transition: border-color 0.25s;
  width: 100%;
  margin: 10px auto;
}

.c2::-webkit-input-placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c2::-moz-placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c2:-ms-input-placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c2::placeholder {
  color: #cccccc;
  text-overflow: ellipsis;
  line-height: normal;
}

.c2:hover,
.c2:focus,
.c2.focused {
  border-color: #6e1fff;
  outline: 0;
}

.c2:disabled {
  border: 2px dotted #e5e5e5;
}

.c1 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
}

.c3 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 12px;
  line-height: 1.4em;
  margin-top: 0;
  color: #f7625a;
  visibility: hidden;
  -webkit-transition: visibility 0.2s linear;
  transition: visibility 0.2s linear;
  -webkit-animation: 0.25s linear;
  animation: 0.25s linear;
  height: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Title
  </div>
  <input
    className="c2"
    placeholder="Placeholder"
  />
  <div
    className="c3"
  />
</div>
`;
